pkg/cloudstorage/manager.go

```diff
@@ -0,0 +1,35 @@
+package cloudstorage
+
+import (
+	corev1alpha1 "github.com/libopenstorage/operator/pkg/apis/core/v1alpha1"
+)
+
+// CloudDriveConfig is the configuration for a single drive
+type CloudDriveConfig struct {
+	// Type of cloud storage
+	Type string `json:"type,omitempty"`
+	// Size of cloud storage
+	SizeInGiB uint64 `json:"sizeInGiB,omitempty"`
+	// IOPS provided by cloud storage
+	IOPS uint32 `json:"iops,omitempty"`
+	// Options are additional options to the storage
+	Options map[string]string `json:"options,omitempty"`
+}
+
+// Config is the cloud storage configuration for a node
+// TODO: This struct should eventually move under StorageNodeStatus
+type Config struct {
+	// CloudStorage is the current cloud configuration
+	CloudStorage []CloudDriveConfig `json:"cloudStorage,omitempty"`
+	// StorageInstancesPerZone is the number of storage instances per zone
+	StorageInstancesPerZone int `json:"storageInstancesPerZone,omitempty"`
+}
+
+// Manager provides an interface to interact with cloud storage
+// provisioner. It is an abstraction layer to interact with the APIs in
+// libopenstorage/cloudops repository
+type Manager interface {
```

Comment:
Why not have this in the storage driver? We can have a Null implementation or not implemented error for other drivers who do not want to implement it.
It will be easier to call `driver.GetStorageNodeConfig()` from the controller in that case. If implemented we use it else we don't.

Reply:
Wouldn't that cause too much of back and forth between the driver and controller?

Currently the controller just calls one function GetStoragePodSpec to get the whole template. Instead the controller will have to call
- GetStorageNodeConfig and get the drive configs
- pass them back to the storage driver to GetStoragePodSpec to generate the spec. 

With this method the controller does not need to know what cloud storage and storage distribution is and the logic relies on the storage driver.

Reply:
Got it. I thought this was going to be called from the controller. If it's being called from the driver, then it's fine.